### PR TITLE
fix(bigquery_usage ingestion): add partition decorator to regex, move exception handling to after matching, add table snapshots

### PR DESCRIPTION
**Pull request by courtney-lang**
## Summary
changes to bigquery_usage ingestion:
- fix: add partition decorator to partitioned table regex
- refactor: move invalid table exception handling to after regex matching
- feat: add bigquery table snapshot handling

#### fix(ingestion): add partition decorator to partitioned table regex
- Add partition decorator `$` to regex 
- Documentation: https://cloud.google.com/bigquery/docs/managing-partitioned-table-data

#### refactor(ingestion): move invalid table exception handling to after regex matching
- Check for partitioned table matching (and table snapshot matching) before handling exceptions that don't match

#### feat(ingestion): add bigquery table snapshot handling
- Add bigquery table snapshot regex and matching
- Documentation: https://cloud.google.com/bigquery/docs/table-snapshots-intro

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by rslanka** [comment]
LGTM!

File: metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py [hunk 1]
```diff
@@ -37,7 +37,11 @@
 
 # Handle yearly, monthly, daily, or hourly partitioning.
 # See https://cloud.google.com/bigquery/docs/partitioned-tables.
-PARTITIONED_TABLE_REGEX = re.compile(r"^(.+)_(\d{4}|\d{6}|\d{8}|\d{10})$")
+PARTITIONED_TABLE_REGEX = re.compile(r"^(.+)\$(\d{4}|\d{6}|\d{8}|\d{10})$")
+
+# Handle table snapshots
+# See https://cloud.google.com/bigquery/docs/table-snapshots-intro. 
+SNAPSHOT_TABLE_REGEX = re.compile(r"^(.+)@(\d{13})$")
```
[the remark this answers is not in the document]
Comment (courtney-lang) [reply metadata-ingestion/src/datahub/ingestion/source/usage/bigquery_usage.py:44]: @rslanka - Yes, when BigQuery creates the snapshot, it is using internal time travel properties: https://cloud.google.com/bigquery/docs/time-travel

**Comment by shirshanka** [comment]
Hi @courtney-lang, thanks for the contrib!

1. Could you run `./gradlew :metadata-ingestion:lintFix` to fix the formatting issues that flake is complaining about? 
2. You might want to rebase your branch since CI on your branch might be impacted by a urllib related bug that has been fixed upstream. 

thanks!

**Comment by courtney-lang** [comment]
@shirshanka - all set, hopefully I did the rebase correctly 🤞

**Review by shirshanka — APPROVED** [review]
LGTM.

**Pull request merged**